paddle/fluid/operators/mkldnn/slice_mkldnn_op.cc

```diff
@@ -79,11 +95,27 @@ class SliceMKLDNNKernel : public framework::OpKernel<T> {
         reorder_handler.AcquireReorder(reorder_dst_memory_p, slice_mem_p);
     auto& astream = platform::MKLDNNDeviceContext::tls().get_stream();
     reorder_p->execute(astream, *slice_mem_p, *reorder_dst_memory_p);
+
+    std::vector<int64_t> new_out_dims(slice_dims.size() - decrease_axis.size());
+
+    for (size_t i = 0; i < decrease_axis.size(); ++i) {
+      slice_dims[decrease_axis[i]] = 0;
+    }
+
+    int j = 0;
+    for (size_t i = 0; i < slice_dims.size(); ++i) {
+      if (slice_dims[i] != 0) new_out_dims[j++] = slice_dims[i];
+    }
+
+    // paddle doesn't support rank 0 tensors, instead we use [1]
+    if (new_out_dims.size() == 0) new_out_dims.emplace_back(1);
```

Comment:
You can use a range based for loop and skip computation for empty output dims:
```suggestion
    std::vector<int64_t> new_out_dims(slice_dims.size() - decrease_axis.size());

    // paddle doesn't support rank 0 tensors, instead we use [1]
    if (new_out_dims.size() == 0) new_out_dims.emplace_back(1);
    else {
        for (const auto& axis: decrease_axis) {
          slice_dims.at(axis) = 0;
        }

        int i = 0;
        for (const auto& slice_dim: slice_dims) {
          if (slice_dim != 0) new_out_dims.at(i++) = slice_dim;
        }
    }
```